### PR TITLE
droplet: Add with_droplet_agent attribute for create requests.

### PR DESCRIPTION
**Pull request by andrewsomething**

### CODE REVIEW

File: specification/resources/droplets/models/droplet_create.yml [hunk 1]
```diff
@@ -93,6 +93,16 @@ properties:
...
 
+  with_droplet_agent:
+    type: boolean
+    example: true
+    description: A boolean indicating whether to install the DigitalOcean agent
+      used for providing access to the Droplet web console in the control panel.
+      By default, the agent is installed on new Droplets but installation errors
+      are ignored. To prevent it from being installed, set to `false`. To make
+      installation errors fatal, failing the create request, explicitly set it
+      to `true`.
```
Comment (andrewsomething) [inline specification/resources/droplets/models/droplet_create.yml:104]: @house-lee Can you confirm that I am correctly describing the logic here?

Comment (house-lee) [reply specification/resources/droplets/models/droplet_create.yml:104]: It looks good to me, but to be exact, it's only affecting how we react to incompatibility (i.e. OS not supported), we (as the droplet service) can't predict if there will be any installation error at this point. So not sure if you want to reveal that detail to customers, but I'll leave it up to you.

Comment (andrewsomething) [reply specification/resources/droplets/models/droplet_create.yml:104]: :+1:  Added "OS not supported" as an example of the type of error we're talking about.

**Review by house-lee — COMMENTED** [review]
👍

**Review by ChiefMateStarbuck — APPROVED** [review]

**Pull request merged**